### PR TITLE
feat(cli): add the cluster-list-example (#15690)

### PR DESCRIPTION
**Pull request by gaurangkudale**
add the cluster-list-example for improvement for #15690

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.

### CODE REVIEW

File: docs/user-guide/commands/argocd_cluster_list.md [hunk 1]
```diff
@@ -8,6 +8,24 @@ List configured clusters
...
+argocd cluster list
+
+# List Cluster via specifing the server
+argocd cluster list --server <ARGOCD_SERVER_ADDRESS>
+
+# List Clusters in JSON Format
+argocd cluster list -o json --server <ARGOCD_SERVER_ADDRESS>
+
+# List Clusters in YAML Format
+argocd cluster list -o yaml --server <ARGOCD_SERVER_ADDRESS>
```
Comment (ishitasequeira) [inline docs/user-guide/commands/argocd_cluster_list.md:25]: Can you also add an example for `-o server`?

**Review by ishitasequeira — CHANGES_REQUESTED** [review]

File: docs/user-guide/commands/argocd_cluster_list.md (hunk 1 above)
Comment (gaurangkudale) [reply docs/user-guide/commands/argocd_cluster_list.md:25]: Yes 
I have added the example

**Comment by gaurangkudale** [comment]
Hi @ishitasequeira 
I have done the changes

**Comment by gaurangkudale** [comment]
Hi @crenshaw-dev 
In your free time Can you please review and merge it

**Review by ishitasequeira — APPROVED** [review]
Thanks @gaurangkudale. LGTM!

**Pull request merged**